sync/class.jetpack-sync-module-posts.php

```diff
@@ -175,6 +180,20 @@ function filter_post_content_and_add_links( $post_object ) {
 		return $post;
 	}
 
+	public function save_published( $new_status, $old_status, $post ) {
+		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
+			$this->just_published = $post;
+		}
+	}
+
+	public function send_published( $post_ID, $post, $update ) {
+		if ( $this->just_published === $post ) {
+			$this->just_published = null;
+			$meta = apply_filters( 'jetpack_published_post_meta', array() );
+			do_action( 'jetpack_published_post', $post_ID, $post, $update, $meta );
```

Comment:
Do we really need to send the whole post object as well as the $update post object? 
I think $meta and $post_id should be enough.